File: parcel-locker-react/src/Service/ParcelService.js
import axios from 'axios';
import authHeader from './AuthHeader';
import Cookies from 'js-cookie';
import AuthService from '../Service/AuthService';

const API_URL = "http://localhost:8080/parcelhandler/parcel/";


//Csomag küldése a weblapról feladási kóddal
//Ez még csak egy előzetes csomagfeladás. A felhasználó megkapja email-ben a csomagfeladási kódot
//A végleges csomagfeladás az automatánál történik
//Jwt token szükséges
const sendParcelWithCodeFromWebpage = (formValues) => {
    return axios.post(API_URL + "sendparcelwithcodefromwebpage", formValues,
        { headers: authHeader() });
};

//Csomag követése
const followParcel = (uniqueParcelId) => {
    return axios.get(API_URL + "followparcel/" + uniqueParcelId);
}

//Felhasználó csomagjainak lekérése
//Típusok:
//all - összes csomag
//reserved - online feladott, automatában még nem elhelyezett csomagok
//notPickedUp - még át nem vett csomagok. Szállítás alatti csomagok
//pickedUp - átvett csomagok. Sikeresen lezárt küldések
//Jwt token szükséges
//User szerepkör szükséges
const getParcelsOfUser = (type) => {
    return axios.get(API_URL + "getparcelsofuser/" + AuthService.getCurrentUser().emailAddress + "/" + type,
        { headers: authHeader() });
}

//Csomag törlése
//Felhasználó kitörli az előzetes csomagfeladást
//Jwt token szükséges
//User szerepkör szükséges
const deleteMyParcel = (parcelId) => {
    return axios.delete(API_URL + "deletemyparcel/" + parcelId,
        { headers: authHeader() });
}

//Központi raktárak csomagjainak lekérése
//Jwt token szükséges
//Admin szerepkör szükséges
const getParcelsOfStore = (storeId) => {
    return axios.get(API_URL + "getparcelsofstore/" + storeId,
        { headers: authHeader() });
}

//Csomag átvételi ideje lejárt, ezért az a központi raktárban van
//Csomag újraindítása az automatához
//pickingUpExpired mező módosítása. True vagy false
//Jwt token szükséges
//Admin szerepkör szükséges
const updatePickingUpExpired = (parcelId) => {
    return axios.get(API_URL + "updatepickingupexpired/" + parcelId,
        { headers: authHeader() });
}

//Futárok csomagjainak lekérése
//Jwt token szükséges
//Admin szerepkör szükséges
const getParcelsOfCourier = (courierId) => {
    return axios.get(API_URL + "getparcelsofcourier/" + courierId,
        { headers: authHeader() });
}

//Automaták csomagjainak lekérése
//Jwt token szükséges
//Admin szerepkör szükséges
const getParcelsOfParcelLocker = (parcelLockerId) => {
    return axios.get(API_URL + "getparcelsofparcellocker/" + parcelLockerId,
        { headers: authHeader() });
}

//Csomagátvételi lejárati idő meghosszabbítása
//Jwt token szükséges
//Admin szerepkör szükséges
const updatePickingUpExpirationDate = (parcelId, newDate) => {
    return axios.get(API_URL + "updatepickingupexpirationdate/" + parcelId + "/" + newDate,
        { headers: authHeader() });
}

//Csomagfeladási lejárati idő meghosszabbítása
//Jwt token szükséges
//Admin szerepkör szükséges
const updateSendingExpirationDate = (parcelId, newDate) => {
    return axios.get(API_URL + "updatesendingexpirationdate/" + parcelId + "/" + newDate,
        { headers: authHeader() });
}

const ParcelService = {
    sendParcelWithCodeFromWebpage,
    followParcel,
    getParcelsOfUser,
    deleteMyParcel,
    getParcelsOfStore,
    updatePickingUpExpired,
    getParcelsOfCourier,
    getParcelsOfParcelLocker,
    updatePickingUpExpirationDate,
    updateSendingExpirationDate

};

export default ParcelService;

File: parcel-locker-android/app/src/main/java/com/example/parcel_locker_android/service/ParcelService.java
package com.example.parcel_locker_android.service;

import com.example.parcel_locker_android.payload.CurrentUser;
import com.example.parcel_locker_android.payload.ParcelDTO;
import com.example.parcel_locker_android.payload.request.SendParcelWithCodeFromWebpageRequest;
import com.example.parcel_locker_android.payload.response.FollowParcelResponse;
import com.example.parcel_locker_android.payload.response.StringResponse;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ParcelService {

    String API_URL = "parcelhandler/parcel/";

    //Csomag leadása a központi raktárba
    //Jwt token szükséges
    @GET(API_URL + "handparceltostore/{uniqueCourierId}/{uniqueParcelId}")
    Call<StringResponse> handParcelToStore(@Path("uniqueCourierId") String uniqueCourierId,
                                           @Path("uniqueParcelId") String uniqueParcelId,
                                           @Header("Authorization") String jwtToken);

    //Csomag felvétele a központi raktárból
    //Jwt token szükséges
    @GET(API_URL + "pickupparcelfromstore/{uniqueCourierId}/{uniqueParcelId}")
    Call<StringResponse> pickUpParcelFromStore(@Path("uniqueCourierId") String uniqueCourierId,
                                           @Path("uniqueParcelId") String uniqueParcelId,
                                           @Header("Authorization") String jwtToken);

    //Csomag nyomonkövetése
    //Jwt token nem szükséges
    @GET(API_URL + "followparcel/{uniqueParcelId}")
    Call<FollowParcelResponse> followParcel(@Path("uniqueParcelId") String uniqueParcelId);

    //Csomag küldése a weblapról feladási kóddal
    //Ez még csak egy előzetes csomagfeladás. A felhasználó megkapja email-ben a csomagfeladási kódot
    //A végleges csomagfeladás az automatánál történik
    //Jwt token szükséges
    @POST(API_URL + "sendparcelwithcodefromwebpage")
    Call<StringResponse> sendParcelWithCodeFromWebpage(@Body SendParcelWithCodeFromWebpageRequest request,
                                                       @Header("Authorization") String jwtToken);

    //Felhasználó csomagjainak lekérése
    //Típusok:
    //all - összes csomag
    //reserved - online feladott, automatában még nem elhelyezett csomagok
    //notPickedUp - még át nem vett csomagok. Szállítás alatti csomagok
    //pickedUp - átvett csomagok. Sikeresen lezárt küldések
    //Jwt token szükséges
    //User szerepkör szükséges
    @GET(API_URL + "getparcelsofuser/{emailAddress}/{type}")
    Call<List<ParcelDTO>> getParcelsOfUser(@Path("emailAddress") String emailAddress,
                                @Path("type") String type,
                                @Header("Authorization") String jwtToken);

    //Csomag törlése
    //Felhasználó kitörli az előzetes csomagfeladást
    //Jwt token szükséges
    //User szerepkör szükséges
    @DELETE(API_URL + "deletemyparcel/{parcelId}")
    Call<StringResponse> deleteMyParcel(@Path("parcelId") Long parcelId,
                                           @Header("Authorization") String jwtToken);




}
